api/prisma/schema.prisma

```diff
@@ -0,0 +1,144 @@
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "mongodb"
+  url      = env("MONGOHQ_URL")
+}
+
+type DonationStartDate {
+  date DateTime @map("_date") @db.Date
+  when String   @map("_when")
+}
+
+type UserBadges {
+  /// Could not determine type: the field only had null or empty values in the sample set.
+  coreTeam Json?
+}
+
+type UserCompletedChallenges {
+  completedDate Float
+  id            String
+}
+
+type UserProfileUi {
+  isLocked      Boolean
+  showAbout     Boolean
+  showCerts     Boolean
+  showDonation  Boolean
+  showHeatMap   Boolean
+  showLocation  Boolean
+  showName      Boolean
+  showPoints    Boolean
+  showPortfolio Boolean
+  showTimeLine  Boolean
+}
+
+model AccessToken {
+  id      String   @id @map("_id")
+  created DateTime @db.Date
+  /// Multiple data types found: Float: 70.3%, Int: 29.7% out of 118 sampled entries
+  ttl     Json
+  userId  String   @db.ObjectId
+}
+
+model AuthToken {
+  id      String   @id @map("_id")
+  created DateTime @db.Date
+  ttl     Int
+  userId  String   @db.ObjectId
+}
+
+model Donation {
+  id             String            @id @default(auto()) @map("_id") @db.ObjectId
+  amount         Int
+  customerId     String
+  duration       String
+  email          String
+  provider       String
+  startDate      DonationStartDate
+  subscriptionId String
+  userId         String            @db.ObjectId
+}
+
+model UserToken {
+  id      String   @id @map("_id")
+  created DateTime @db.Date
+  ttl     Float
+  userId  String   @db.ObjectId
+}
+
+model WebhookToken {
+  id      String   @id @map("_id")
+  created DateTime @db.Date
+  ttl     Float
+  userId  String   @db.ObjectId
+}
+
+model expressRateRecords {
+  id             String @id @default(auto()) @map("_id") @db.ObjectId
+  /// Field referred in an index, but found no data to define the type.
+  expirationDate Json?
+
+  @@index([expirationDate], map: "expirationDate_1")
+}
```

Comment:
1) Does this have to do with our rate limiting?
2) If so, do we store this in a collection on MongoDB?
3) If so, should we not be using a volatile storage option for this?
4) This is not a permanent ban, is it?

Reply:
1. Yes
2. Yes
3. That would make more sense, I think.
4. Nope, hence the expiration date.

Reply:
I will create a pin for the new api